### ISSUE TITLE
Create website listing all the implemented lints and their info

### ISSUE THREAD

**Issue by obi1kenobi** [obi1kenobi/cargo-semver-checks#819]
Originally suggested here:
https://rust-lang.zulipchat.com/#narrow/stream/421156-gsoc/topic/Project.3A.20Adding.20lint.20configuration.20to.20cargo-semver-checks/near/452638414

> Any plans for generating a webpage of lints like [crates.io](http://crates.io/) ([example](https://rust-lang.github.io/rust-clippy/master/index.html#/print_stderr))? I know @Scott Schafer is looking into this for Cargo.

This is a good idea! We need to look into what it would take to set this up and keep it maintained.

**Comment by epage** [comment]
rustc and clippy generate their documentation and Cargo is looking to do the same.  No one hand-maintains the webpage.

iirc clippy scrapes the code and generates it.  Since Cargo isn't using a macro (yet?) for defining lints, I suggested to @Muscraft that we have a test or an xtask that calls into cargo-the-library to read the lints.

For you, the lints are data-driven so you can just load all of the files.

For generating, I think it would be good to see if a library could be made from clippy to accept a lint definition and generate it, so we all can share the same generator.

**Comment by obi1kenobi** [comment]
Sharing the generator would be ideal! It would be pretty easy to parse the lints' information and plumb that through to the generator to make the website.

**Comment by Muscraft** [comment]
[Cargo's lint docs](https://doc.rust-lang.org/nightly/cargo/reference/lints.html) currently get [generated by an xtask](https://github.com/rust-lang/cargo/blob/5f6b9a92201d78af75dc24f14662c3e2dacbbbe1/crates/xtask-lint-docs/src/main.rs) that reads fields from our `Lint` type. It works reasonably well currently, but examples (and their output) have to be manually updated. In the future, I plan on using Cargo's UI tests for examples. To ensure the docs get updated, the output gets [checked in CI](https://github.com/rust-lang/cargo/blob/5f6b9a92201d78af75dc24f14662c3e2dacbbbe1/.github/workflows/main.yml#L86-L91).

**Comment by obi1kenobi** [comment]
Thanks for the links, super useful!

**Comment by SuperSonicHub1** [comment]
A pragmatic suggestion: we could store lint documentation using doc comments. This would allow users to read lint docs offline using `cargo doc`, host them easily on docs.rs, and keep the documentation close to code.

**Comment by obi1kenobi** [comment]
I don't think that would work, @SuperSonicHub1. Our lints are data, not types or functions, so there's no Rust item to which to attach the doc comment for each lint.

In clippy, each lint has its own type. Here, we have one type for all lints, and we deserialize the lints themselves out of RON files into instances of that struct.
